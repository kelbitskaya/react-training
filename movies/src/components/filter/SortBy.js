import React, {useCallback} from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { fetchMovies } from '../../store/actions/actions';
import Dropdown from '../common/dropdown/Dropdown';
import Constants from '../constants';

const matchDispatchToProps = {updateMovies: fetchMovies};

const SortBy = (props) =>  {
  const { updateMovies } = props;

  const applySortMovies = useCallback((sortType) => {
    updateMovies(sortType.type)
  }, [updateMovies]);

  return (
    <div className="sort-by">
      <p className="sort-by__text">sort by</p>
      <Dropdown options={Constants.FILTERS} updateMovies={applySortMovies}/>
    </div>
  );
};

SortBy.propTypes = {
  updateMovies: PropTypes.func.isRequired,
};


export default connect(null, matchDispatchToProps)(SortBy);
