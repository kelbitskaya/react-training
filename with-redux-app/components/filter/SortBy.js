import React, {useCallback} from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { useHistory, useLocation } from "react-router-dom";
import { fetchMovies } from '../../store/actions/actions';
import Dropdown from '../common/dropdown/Dropdown';
import Constants from '../constants';
import {useRouter} from "next/router";

const matchDispatchToProps = {updateMovies: fetchMovies};

const SortBy = (props) =>  {
  const { updateMovies } = props;

  // const history = useHistory();
  // const location = useLocation();

  const router = useRouter();
  const { pathname } = router;

  const applySortMovies = useCallback((sortType) => {
    // const currentLocation = !!pathname.search ? pathname.search : router.location.search;
    // const params = new URLSearchParams(currentLocation);
    // const title = params.get('q');
    // const filter = params.get('filter');
    //
    // updateMovies(sortType.type, filter, title);
    // history.push(`/search?${params.toString()}`);
  }, [updateMovies]);

  return (
    <div className="sort-by">
      <p className="sort-by__text">sort by</p>
      <Dropdown options={Constants.FILTERS} updateMovies={applySortMovies}/>
    </div>
  );
};

SortBy.propTypes = {
  updateMovies: PropTypes.func.isRequired,
};


export default connect(null, matchDispatchToProps)(SortBy);
