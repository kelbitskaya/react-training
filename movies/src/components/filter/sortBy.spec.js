import React from 'react';
import SortBy from './SortBy';
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import rerender from "react-test-renderer";
import routeData from "react-router";

const mockLocation = {
  pathname: '/welcome',
  hash: '',
  search: '',
  state: ''
};

describe("SortBy component", () => {
  beforeEach(()=>{
    jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation)
  });

  test("should render SortBy component", ()=> {
    const initialStore = {};
    const store = configureStore([])(initialStore);
    const component = rerender.create(
      <Provider store={store}>
        <SortBy/>
      </Provider>
      );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });


  test('SortBy has dropdown ', () => {
    const initialStore = {};
    const store = configureStore([])(initialStore);
    const component = mount(
      <Provider store={store}>
        <SortBy/>
      </Provider>);

    const button = component.find(".dropdown");
    expect(button.length).toBe(1);
  });

});
