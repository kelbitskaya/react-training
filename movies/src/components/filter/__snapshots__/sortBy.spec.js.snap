// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultFilter component should render ResultFilter component 1`] = `
<div
  className="sort-by"
>
  <p
    className="sort-by__text"
  >
    sort by
  </p>
  <div
    className="dropdown"
  >
    <button
      className="dropbtn "
      onClick={[Function]}
      type="button"
    >
      release date
    </button>
    <ul
      className="dropdown-content "
    >
      <li
        className="dropdown-content__item"
        datatype="release_date"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        type="button"
      >
        release date
      </li>
      <li
        className="dropdown-content__item"
        datatype="title"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        type="button"
      >
        title
      </li>
      <li
        className="dropdown-content__item"
        datatype="rating"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        type="button"
      >
        rating
      </li>
    </ul>
  </div>
</div>
`;

exports[`SortBy component should render SortBy component 1`] = `
<div
  className="sort-by"
>
  <p
    className="sort-by__text"
  >
    sort by
  </p>
  <div
    className="dropdown"
  >
    <button
      className="dropbtn "
      onClick={[Function]}
      type="button"
    >
      release date
    </button>
    <ul
      className="dropdown-content "
    >
      <li
        className="dropdown-content__item"
        datatype="release_date"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        type="button"
      >
        release date
      </li>
      <li
        className="dropdown-content__item"
        datatype="title"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        type="button"
      >
        title
      </li>
      <li
        className="dropdown-content__item"
        datatype="rating"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        type="button"
      >
        rating
      </li>
    </ul>
  </div>
</div>
`;
